### PR TITLE
feat: added lastLookConfig to transaction entity

### PR DESCRIPTION
**Pull request by alexkroeger**
# Description

For RFQM the transaction worker will have to know to ping the market maker before submission. The details of how to do so are stored in `lastLookConfig`.

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

**Review by phil-ociraptor — APPROVED** [review]

**Review by PirosB3 — COMMENTED** [review]
- Do we need to add a new `status` field that allows us to determine if an order is pending, rejected by a MM, or complete?
- Do we need to also have a field `order_hash` alongside `tx_hash`? it looks like we would use both fields to retreive an order

**Comment by alexkroeger** [comment]
> * Do we need to add a new `status` field on the table itself (not json) that allows us to determine if an order is pending, rejected by a MM, or complete?
> * Do we need to also have a field `order_hash` alongside `tx_hash`? it looks like we would use both fields to retreive an order

There is already a `status` field to determine where in the lifecycle the tx is. There is an enum type for it, I added an option for `lastLookRejected`, which will end the submission process.
`order_hash` shouldn't be necessary to track in a separate field. It's in the lastLookConfig if we want to expose it.

File: test/rfqm_test.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    takerToken: '0xcd',
+                    txOrigin: '0x00',
+                },
+            };
+            const txEntityOpts: TransactionEntityOpts = {
+                refHash: '0x00',
+                apiKey: 'averycoolkey',
+                txHash: '0x00',
+                takerAddress: '0x00',
+                status: 'submitted',
```
[the remark this answers is not in the document]
Comment (alexkroeger) [reply test/rfqm_test.ts:56]: it is! in types

File: src/types.ts [hunk 2]
```diff
@@ -714,4 +714,19 @@ export interface BucketedPriceDepth {
     bucket: number;
     bucketTotal: BigNumber;
 }
+
+export interface LastLookConfig {
+    makerUri: string;
```
[the remark this answers is not in the document]
Comment (alexkroeger) [reply src/types.ts:442]: I've seen both unfortunately in asset-swapper. I prefer URL here, going to change

**Pull request closed without merging**